File: facebook/src/test/java/com/facebook/internal/JsonUtilTest.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.internal;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

import com.facebook.FacebookTestCase;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map.Entry;
import java.util.Set;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;

public final class JsonUtilTest extends FacebookTestCase {

  @Test
  public void testJsonObjectClear() throws JSONException {
    JSONObject jsonObject = new JSONObject();

    jsonObject.put("hello", "world");
    jsonObject.put("hocus", "pocus");

    JsonUtil.jsonObjectClear(jsonObject);
    assertEquals(0, jsonObject.length());
  }

  @Test
  public void testJsonObjectContainsValue() throws JSONException {
    JSONObject jsonObject = new JSONObject();

    jsonObject.put("hello", "world");
    jsonObject.put("hocus", "pocus");

    assertTrue(JsonUtil.jsonObjectContainsValue(jsonObject, "pocus"));
    assertFalse(JsonUtil.jsonObjectContainsValue(jsonObject, "Fred"));
  }

  @Test
  public void testJsonObjectEntrySet() throws JSONException {
    JSONObject jsonObject = new JSONObject();

    jsonObject.put("hello", "world");
    jsonObject.put("hocus", "pocus");

    Set<Entry<String, Object>> entrySet = JsonUtil.jsonObjectEntrySet(jsonObject);
    assertEquals(2, entrySet.size());
  }

  @Test
  public void testJsonObjectKeySet() throws JSONException {
    JSONObject jsonObject = new JSONObject();

    jsonObject.put("hello", "world");
    jsonObject.put("hocus", "pocus");

    Set<String> keySet = JsonUtil.jsonObjectKeySet(jsonObject);
    assertEquals(2, keySet.size());
    assertTrue(keySet.contains("hello"));
    assertFalse(keySet.contains("world"));
  }

  @Test
  public void testJsonObjectPutAll() throws JSONException {
    HashMap<String, Object> map = new HashMap<String, Object>();
    map.put("hello", "world");
    map.put("hocus", "pocus");

    JSONObject jsonObject = new JSONObject();
    JsonUtil.jsonObjectPutAll(jsonObject, map);

    assertEquals("pocus", jsonObject.get("hocus"));
    assertEquals(2, jsonObject.length());
  }

  @Test
  public void testJsonObjectValues() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("hello", "world");
    jsonObject.put("hocus", "pocus");

    Collection<Object> values = JsonUtil.jsonObjectValues(jsonObject);

    assertEquals(2, values.size());
    assertTrue(values.contains("world"));
  }
}

File: facebook/src/test/java/com/facebook/internal/JsonUtilTest.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.internal

import com.facebook.FacebookTestCase
import java.util.HashMap
import junit.framework.Assert
import org.json.JSONObject
import org.junit.Test

class JsonUtilTest : FacebookTestCase() {

  @Test
  fun testJsonObjectClear() {
    val jsonObject = JSONObject()
    jsonObject.put("hello", "world")
    jsonObject.put("hocus", "pocus")
    JsonUtil.jsonObjectClear(jsonObject)
    Assert.assertEquals(0, jsonObject.length())
  }

  @Test
  fun testJsonObjectContainsValue() {
    val jsonObject = JSONObject()
    jsonObject.put("hello", "world")
    jsonObject.put("hocus", "pocus")
    Assert.assertTrue(JsonUtil.jsonObjectContainsValue(jsonObject, "pocus"))
    Assert.assertFalse(JsonUtil.jsonObjectContainsValue(jsonObject, "Fred"))
  }

  @Test
  fun testJsonObjectEntrySet() {
    val jsonObject = JSONObject()
    jsonObject.put("hello", "world")
    jsonObject.put("hocus", "pocus")
    val entrySet = JsonUtil.jsonObjectEntrySet(jsonObject)
    Assert.assertEquals(2, entrySet.size)
  }

  @Test
  fun testJsonObjectKeySet() {
    val jsonObject = JSONObject()
    jsonObject.put("hello", "world")
    jsonObject.put("hocus", "pocus")
    val keySet = JsonUtil.jsonObjectKeySet(jsonObject)
    Assert.assertEquals(2, keySet.size)
    Assert.assertTrue(keySet.contains("hello"))
    Assert.assertFalse(keySet.contains("world"))
  }

  @Test
  fun testJsonObjectPutAll() {
    val map = HashMap<String, Any>()
    map["hello"] = "world"
    map["hocus"] = "pocus"
    val jsonObject = JSONObject()
    JsonUtil.jsonObjectPutAll(jsonObject, map)
    Assert.assertEquals("pocus", jsonObject["hocus"])
    Assert.assertEquals(2, jsonObject.length())
  }

  @Test
  fun testJsonObjectValues() {
    val jsonObject = JSONObject()
    jsonObject.put("hello", "world")
    jsonObject.put("hocus", "pocus")
    val values = JsonUtil.jsonObjectValues(jsonObject)
    Assert.assertEquals(2, values.size)
    Assert.assertTrue(values.contains("world"))
  }
}
